pages/dashboard.js

```diff
@@ -146,10 +156,50 @@ export default function Dashboard(props) {
                 orderedActiveAgreement.push(benefit)
                 break
             }
+
+            //determine what benefits to advertise
```

Comment:
there is a fair bit of repetitive code here, see if there are ways to reduce the repetitiveness so it's easy to expand on.

Reply:
Tried a different approach, haven't added tests yet though. Not sure about my approach for verifying program codes since all it does is verify that a benefit has a correct code, but I don't think there's an alternative approach that verifies that the  program code matches with the typecode without un-generalizing it. 

Side note, I noticed that atm CPPD benefits all fall under the type of `CPPRetirement` which, assuming it isn't intentional, means that CPPD advertisement cards will continue to show even if there's CPPD benefit cards present.

Reply:
Some of the separation of concerns is starting to show with this, which is great, but more can be done to make the code more readable.
Also, I think you've started to over complicated the logic, it should pretty easy to follow - the logic itself isn't that complicated. Each application card has it's own matching conditions based on Program, sometimes Type and sometimes Status. While each condition is unique, the way we've set up the benefit objects is so they are not unique, so use that!